internal/service/mongodb/mongodb_users.go

```diff
@@ -340,10 +344,6 @@ func GetMongoDbUserList(ctx context.Context, config *conn.ProviderConfig, id str
 		}
 	}
 
-	if len(filteredUsers) == 0 {
-		return nil, nil
-	}
-
 	for _, user := range resp.CloudMongoDbUserList {
 		if user != nil && user.UserName != nil {
 			if !containsInUsergList(*user.UserName, filteredUsers) {
```

Comment:
This logic seems weird.
This function `GetMongoDbUserList()` checks if the `users` passed in as arguments are included in the CloudMongoDbUserList.
But in this logic, adding all CloudMongodbUserList regardless of this, so there's no need to create a list of filteredUsers to compare against users above.

Reply:
in case of list, diff occurs if the order of the list in the .tf code and the order of the list in the search result are not the same.

Reply:
we could avoid unnecessary differences caused by the order by using `set` instead of `list`, how about replacing it with `set`?